host/xtest/ffa_spmc_1000.c

```diff
@@ -137,10 +121,12 @@ static void xtest_tee_test_9001(ADBG_Case_t *c)
 	}
 
 	memset(&args, 0, sizeof(args));
-	ADBG_EXPECT_TRUE(c, start_sp_test(endpoint1_id, EP_TEST_SP, &args));
-	ADBG_EXPECT_TRUE(c, check_ok(&args));
+	ADBG_REQUIRE_TRUE(c, exit_test, start_sp_test(endpoint1_id, EP_TEST_SP,
```

Comment:
Usually, when we write tests we try to make the error reporting contain details that give some help to debug the issue. With this, you just report that it failed without any error code.

I would have written this as:
``` C
rc = start_sp_test(endpoint1_id, EP_TEST_SP, &args);
if (!ADBG_EXPECT_COMPARE_SIGNED(c, rc, ==, 0))
     goto out;
```
where start_sp_test() returned the error value instead. Sure, in some cases the error value will not be very useful, but it can never contain less information than `false`. By the way, I avoid the ADBG_REQUIRE_*() macros since they encourage helper macros like your `exit_test`.

Reply:
Ack